File: ANDROID/SMA_v1/lib/models/employeeModel.dart
import 'dart:convert';

class employeeModel {
  final int employeeID;
  final String? employeeCode;
  final int organizationUnitID;
  final String? fullName;
  final int jobTitleID;
  final String? employeeTel;
  final String? employeeEmail;
  final String? employeeImage;
  final String? description;

  employeeModel(
      { required this.employeeID,
        this.employeeCode,
        required this.organizationUnitID,
        this.fullName,
        required this.jobTitleID,
        this.employeeTel,
        this.employeeEmail,
        this.employeeImage,
        this.description,});

  Map<String, dynamic> toJson() {
    return {
      'employeeID': employeeID,
      'employeeCode' : employeeCode,
      'organizationUnitID': organizationUnitID,
      'fullName':fullName,
      'jobTitleID':jobTitleID,
      'employeeTel':employeeTel,
      'employeeEmail':employeeEmail,
      'employeeImage':employeeImage,
      'description':description,

    };
  }

  String modelToJsonEncode(){
    return jsonEncode(toJson());
  }

  factory employeeModel.fromJson(Map<String, dynamic> json) {
    return employeeModel(
        employeeID: json['employeeID'],
        employeeCode : json ['employeeCode'],
        organizationUnitID : json ['organizationUnitID'],
        fullName : json ['fullName'],
        jobTitleID : json ['jobTitleID'],
        employeeTel : json ['employeeTel'],
        employeeEmail: json ['employeeEmail'],
        employeeImage : json ['employeeImage'],
        description : json ['description'],

    );
  }
  @override
  String toString() {
    // TODO: implement toString
    return 'employeeID: $employeeID, employeeCode: $employeeCode, organizationUnitID: $organizationUnitID, fullName: $fullName, jobTitleID: $jobTitleID, employeeTel: $employeeTel, employeeEmail: $employeeEmail, employeeImage: $employeeImage, description: $description';
  }
}

File: API/Model/Models/EmployeeModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Model.Models
{
    public class EmployeeModel
    {
        public int EmployeeID { get; set; }

        public string EmployeeCode { get; set; } = null!;

        public int? OrganizationUnitID { get; set; }

        public string? FullName { get; set; }

        public int? JobTitleID { get; set; }

        public string? EmployeeTel { get; set; }

        public string? EmployeeEmail { get; set; }

        public string? EmployeeImage { get; set; }

        public string? Description { get; set; }
        public string? JobTitleName { get; set; }
        public string? OrganizationUnitName { get; set; }

    }
}
